crmsh/bootstrap.py

```diff
@@ -509,6 +531,45 @@ def _get_sbd_device_from_config(self):
         else:
             return None
 
+    def _restart_cluster_on_needed(self):
+        """
+        Try to configure sbd resource, restart cluster on needed
+        """
+        if not utils.has_resource_running():
+            status("Restarting cluster service")
+            utils.cluster_run_cmd("crm cluster restart")
+            wait_for_cluster()
+            self.configure_sbd_resource()
+        else:
+            warn("To start sbd.service, need to restart cluster service manually on each node")
+            if self.diskless_sbd:
+                warn("Then run \"crm configure property stonith-enabled=true stonith-watchdog-timeout={}\" on any node".format(self._watchdog_timeout))
+            else:
+                self.configure_sbd_resource()
+
+    def _enable_sbd_service(self):
+        """
+        Try to enable sbd service
+        """
+        if _context.cluster_is_running:
+            # in sbd stage, enable sbd.service on cluster wide
+            utils.cluster_run_cmd("systemctl enable sbd.service")
+            self._restart_cluster_on_needed()
+        else:
+            # in init process
+            invoke("systemctl enable sbd.service")
+
+    def _warn_diskless_sbd(self, peer=None):
+        """
+        Give warning when configuring diskless sbd
+        """
+        if (self.diskless_sbd and _context.cluster_is_running) or peer:
+            vote_dict = utils.get_quorum_votes_dict(peer)
+            if int(vote_dict['Expected']) < 2:
+                warn(self.DISKLESS_SBD_WARNING)
+        elif self.diskless_sbd:
+            warn(self.DISKLESS_SBD_WARNING)
```

Comment:
Logic is kind of obscure without knowing the contexts of the callers here, but I don't know how to improve it either :-)

Reply:
Add some comments in this function